ruby_event_store/lib/ruby_event_store/mappers/protobuf.rb

```diff
@@ -39,13 +40,14 @@ def pool
   module Mappers
     class Protobuf
       def initialize(events_class_remapping: {})
+        require_optional_dependency
         @events_class_remapping = events_class_remapping
       end
 
       def event_to_serialized_record(domain_event)
         SerializedRecord.new(
           event_id:   domain_event.event_id,
-          metadata:   YAML.dump(domain_event.metadata),
+          metadata:   ProtobufNestedStruct::HashMapStringValue.dump(domain_event.metadata.each_with_object({}){|(k,v),h| h[k.to_s] =v }),
```

Comment:
So did I get this right:

When we are going _to_  the protobuf world, we need the hash keys to all be of type `String` and that's why we need to call `k.to_s` here?

And when we are in the `serialized_record_to_event` method, we need to do the opposite, i.e. call `k.to_sym` as the `Metadata` class requires keys to be symbols?

Reply:
yep, I don't like it as well. but `Metadata` is already using Symbols so...

Reply:
Yeah, just making sure I followed this logic. Nothing wrong with it, even though it means we have to jump through a few loops here :)